app/controllers/posts_controller.rb

```diff
@@ -41,10 +42,30 @@ def update
         end
     end
 
+    def destroy
+        @post = Post.find(params[:id])   #find the post
+
+        unless current_user == @post.user
+            redirect_to @post, notice: "You are not authorised to delete this post."
+        end
+
+        if @post.soft_delete == nil     #if the post is not deleted, it is equal to nil.
+            @post.soft_delete_f         #then delete it
+            render 'show'
+            #flash.notice = "Your post was deleted."
```

Comment:
I know nothing of Ruby but should this be commented out?

Reply:
I meant to come back to this actually but completely forgot.  Cheers 💯